src/core/main.c

```diff
@@ -1245,9 +1245,9 @@ static void bump_file_max_and_nr_open(void) {
 #endif
 
 #if BUMP_PROC_SYS_FS_FILE_MAX
-        /* I so wanted to use STRINGIFY(ULONG_MAX) here, but alas we can't as glibc/gcc define that as
-         * "(0x7fffffffffffffffL * 2UL + 1UL)". Seriously. 😢 */
-        if (asprintf(&t, "%lu\n", ULONG_MAX) < 0) {
+        /* The maximum the kernel allows for this since 5.2 is LONG_MAX, use that. (Previously thing where
+         * different but the operation would fail silently.) */
```

Comment:
That comment is bit misleading. The maximum was always the same, it's just that before the error was squashed and now its reported. I think it'd be good to reword the comment. Also, typo: "thing were".

Reply:
Na, the error wasn't just squashed. instead the kerne apparently parsed ULONG_MAX as LONG_MAX, either due to saturation, or due to some casting weirdness... I left this vague on purpose, since I didn't want to do code spelunking to figure out the details... but iirc the previous behaviour was that ULONG_MAX meant LONG_MAX, but ULONG_MAX+1 was not parsed at all... hence there's a differene between "maximum accepted" and "maximum configurable". it would accept something larger than was actually used.

Reply:
`file-max` needs to be calculated vie percpu counters. Those percpu counters are `long`s and not `unsigned long`s so it is not possible for `file-max` to ever go up to `unsigned long`. This restriction could technically be lifted but would require a new percpu api and then you also get trouble with how correct everything is going to be etc. pp. So `file-max` was always `long`. It really wasn't documented clearly anywhere though.

Reply:
yes, internally in the kernel it was "signed long" all the time, just the kernel's sysctl parser perviously parsed values in the range LONG_MAX+1…ULONG_MAX as something weird, and now (correctly) doesn't anymore. Previously only ULONG_MAX+1…∞ where refused.

Reply:
Yeah, just providing some more context. :)

Reply:
OK. I admit I looked at the kernel patch briefly, but it's far from obvious.